smartsim/settings/slurmSettings.py

```diff
@@ -88,7 +88,7 @@ def set_nodes(self, nodes: int) -> None:
         """
         self.run_args["nodes"] = int(nodes)
 
-    def make_mpmd(self, srun_settings: SrunSettings) -> None:
+    def make_mpmd(self, srun_settings: RunSettings) -> None:
```

Comment:
Great eye! I need to do another run over all docstrings.

Reply:
After review, I recall that I left these as-is to identify the subclass we really wanted. I can't change the type in the signature or I get an error from mypy about not matching the parent / liskov sub. principle. 

I did realize that I could tighten up the types on the constructors since those aren't shared and I'm adding that, too. I think this change will let me shift these make_mpmd off the base class and use a concrete type